coalib/parsing/ConfParser.py

```diff
@@ -123,6 +123,12 @@ def __parse_lines(self, lines, origin):
                 if key == '':
                     continue
 
+                if key in current_section.contents and keys != []:
```

Comment:
It won't be possible to omit `keys != []` as that is the condition for `allow_appending` to be `False`. The loop will be entered even when `keys` is empty since we don't have any explicit condition that handles `keys` to be empty. CC @Makman2